x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/ConfigInitialNode.java

```diff
@@ -82,83 +88,81 @@
-        try {
-            // it is odd for the `elasticsearch.yml` file to be missing or not be a regular (the node won't start)
-            // but auto configuration should not be concerned with fixing it (by creating the file) and let the node startup fail
-            if (false == Files.exists(ymlPath) || false == Files.isRegularFile(ymlPath, LinkOption.NOFOLLOW_LINKS)) {
-                terminal.println(unexpectedNoopVerbosityLevel(), String.format(Locale.ROOT, "Skipping security auto configuration because" +
-                        " the configuration file [%s] is missing or is not a regular file", ymlPath));
-                throw new UserException(ExitCodes.CONFIG, null);
-            }
-            // If the node's yml configuration is not readable, most probably auto-configuration isn't run under the suitable user
-            if (false == Files.isReadable(ymlPath)) {
-                terminal.println(unexpectedNoopVerbosityLevel(), String.format(Locale.ROOT, "Skipping security auto configuration because" +
-                        " the configuration file [%s] is not readable", ymlPath));
-                throw new UserException(ExitCodes.NOOP, null);
-            }
-            // Inform that auto-configuration will not run if keystore cannot be read.
-            if (Files.exists(keystorePath) && (false == Files.isRegularFile(keystorePath, LinkOption.NOFOLLOW_LINKS) ||
-                    false == Files.isReadable(keystorePath))) {
-                terminal.println(unexpectedNoopVerbosityLevel(), String.format(Locale.ROOT, "Skipping security auto configuration because" +
-                        " the node keystore file [%s] is not a readable regular file", keystorePath));
-                throw new UserException(ExitCodes.NOOP, null);
-            }
-        } catch (UserException e) {
-            if (options.has(strictOption)) {
-                throw e;
-            } else {
-                return; // silent error because we wish the node to start as usual (skip auto config) if the configuration is read-only
-            }
+        // it is odd for the `elasticsearch.yml` file to be missing or not be a regular (the node won't start)
+        // but auto configuration should not be concerned with fixing it (by creating the file) and let the node startup fail
+        if (false == Files.exists(ymlPath) || false == Files.isRegularFile(ymlPath, LinkOption.NOFOLLOW_LINKS)) {
+            terminal.println(Terminal.Verbosity.NORMAL, String.format(Locale.ROOT, "Skipping security auto configuration because" +
+                    " the configuration file [%s] is missing or is not a regular file", ymlPath));
+            throw new UserException(ExitCodes.CONFIG, null);
+        }
+        // If the node's yml configuration is not readable, most probably auto-configuration isn't run under the suitable user
+        if (false == Files.isReadable(ymlPath)) {
```

Comment:
Just to double check, you want a single exit code no matter the error condition?

Reply:
It looked like an opportunity to remove a few lines of code by moving `false == Files.isReadable(ymlPath)` to the check above. This is mostly about less code since we don't differentiate between CONFIG and NOOP, and _not_ about wanting the same exit code ( that would just be a result, not the cause ). If you think that the different error message has informational value or the different exit code can be useful at some point, I'm more than happy to keep this. As said, this is just a nit suggestion based on personal preference, happy eitherway